packages/x-components/src/components/global-x-bus.vue

```diff
@@ -0,0 +1,122 @@
+<script lang="ts">
+  import { Component } from 'vue-property-decorator';
+  import { Subscription } from 'rxjs';
+  import { forEach } from '../utils';
+  import { WireMetadata, XEvent, XEventPayload } from '../wiring';
+  import { NoElement } from './no-element';
+
+  type EventListeners = Record<
+    XEvent,
+    (payload: XEventPayload<XEvent>, metadata?: WireMetadata) => unknown
+  >;
+
+  /**
+   * A component which purpose is to receive callbacks associated with events and handle the
```

Comment:
```
  /**
   * A component which receives a list of listeners, a key-value pair of an {@link XEvent} and a callback,
   * and iterate over them in order to subscribe each of them to the bus.
   *
   * When an XEvent is registered in the {@link XBus} this component keeps track of the callback returns
   *  in order to send it back letter using the  `SnippetCallbackExecuted` event.
   *
   * @remarks
   *
   * @public
   */
```

Something like this?

Reply:
I would suggest something much more simple.
```ts
  /**
   * This component helps subscribing to any {@link XEvent} using Vue listeners API.
   *
   * @public
   */
```